include/sway/tree/view.h

```diff
@@ -194,6 +194,8 @@ struct sway_view_child {
 	const struct sway_view_child_impl *impl;
 
 	struct sway_view *view;
+	struct sway_view_child *parent;
+	list_t *children;
```

Comment:
nit: I think it would be better if this was a `struct wl_list`

Reply:
I had gone by the mantra of "If you think you need a doubly linked list, you're wrong", but it does make child removal code much nicer, so I changed it.